safe_core/src/client/mock/vault.rs

```diff
@@ -752,14 +754,424 @@ impl Vault {
+                                self.commit_mutation(requester.name());
+                                self.insert_data(
+                                    id,
+                                    Data::AppendOnly(AppendOnlyDataKind::PubUnseq(adata)),
+                                );
+                                Ok(())
+                            }
+                            _ => Err(Error::NoSuchData),
+                        },
+                        ADataAddress::UnpubSeq { .. } => match data.clone() {
+                            AData::UnpubSeq(mut adata) => {
+                                unwrap!(adata.append_owner(owner));
+                                self.commit_mutation(requester.name());
+                                self.insert_data(
+                                    id,
+                                    Data::AppendOnly(AppendOnlyDataKind::UnpubSeq(adata)),
+                                );
+                                Ok(())
+                            }
+                            _ => Err(Error::NoSuchData),
+                        },
+                        ADataAddress::UnpubUnseq { .. } => match data {
+                            AData::UnpubUnseq(mut adata) => {
+                                unwrap!(adata.append_owner(owner));
+                                self.commit_mutation(requester.name());
+                                self.insert_data(
+                                    id,
+                                    Data::AppendOnly(AppendOnlyDataKind::UnpubUnseq(adata)),
+                                );
+                                Ok(())
+                            }
+                            _ => Err(Error::NoSuchData),
+                        },
+                    });
+                Response::SetADataOwner(res)
+            }
+            Request::GetADataOwners {
+                address,
+                owners_index,
+            } => {
+                let res = self
+                    .get_adata(address, requester_pk, request)
+                    .and_then(move |data| {
+                        let idx = match owners_index {
+                            ADataIndex::FromStart(idx) => idx,
+                            ADataIndex::FromEnd(idx) => (data.owners_index() - idx),
+                        };
+                        match data.get_owners(idx) {
+                            Some(owner) => Ok(owner.clone()),
+                            None => Err(Error::NoSuchEntry),
+                        }
+                    });
+                Response::GetADataOwners(res)
+            }
         };
-
         Ok(Message::Response {
             response,
             message_id,
         })
     }
+    //
+    // ======= Append Only Data =======
+    //
+    pub fn get_adata(
+        &mut self,
+        data: ADataAddress,
+        requester_pk: PublicKey,
+        request: Request,
```

Comment:
This should be
```suggestion
        request: &Request,
```
to prevent unnecessary data copying.